server/command.go

```diff
@@ -181,6 +187,112 @@ func (p *Plugin) executeLinkCommand(args *model.CommandArgs, parameters []string
 		return p.cmdError(args.UserId, args.ChannelId, "Unable to create new link.")
 	}
 
+	if p.getConfiguration().SyncUsers > 0 {
+		batchSize := 50
+		go func() {
```

Comment:
For my taste this section of the function is complex enough to have its own function, actually, even if you move that to another function, it ends up being complex enough to split it again. Lets start splitting it in some chunks that makes sense, and lets see if that improve the readability or if from there emerges any interesting abstraction.